xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/uicomponents/job/job.js

```diff
@@ -32,6 +32,12 @@ require.config({
 
 require(['jquery', 'xwiki-meta', 'JobRunner'], function($, xm, JobRunner) {
   'use strict';
+  // prevent the user from leaving the job
+  $(window).on("beforeunload", function(e) {
+    e.preventDefault();
+    e.returnValue = "";
+  });
```

Comment:
Same as above. ``job.js`` can be loaded anytime before, during or after the job is finished and there might be no question to be asked or the job could be run in non-interactive mode.